source/options.js

```diff
@@ -30,3 +30,60 @@ const update = async ({target: input}) => {
 for (const input of document.querySelectorAll('#options-form [name]')) {
 	input.addEventListener('change', update);
 }
+
+const button = document.querySelector('.js-auth-flow');
+button.addEventListener('click', async () => {
+	const options = await syncStore.getAll();
+	const url = new URL(options.rootUrl);
+	const origin = url.origin;
+
+	try {
+		const alreadyGranted = await queryPermission(origin);
+
+		if (!alreadyGranted) {
+			const granted = await requestPermission(origin);
+
+			if (granted) {
+				const CLIENT_ID = 'c209d287b799e3899f7a';
+				const CLIENT_SECRET = '2eab029032e28d0ca69b2cbf4332c813672f0003';
```

Comment:
Is there a danger in exposing the secret?

Reply:
As far as I can tell GitHub doesn't provide with implicit grant flow. You have to exchange a authorization token for access token.

And for this step you need client secret. These client secrets are usually stored on a proxy handled the access token step. But we do not have that architecture and all apps that have to store client secrets _somehow_ (according to the spec, the spec tells you nothing about storing client secrets).

GitHub however requires you to send the client secret while [auth flow](https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/#2-users-are-redirected-back-to-your-site-by-github) to be sent in the URL parameters 😱🤯, which can be easily intercepted on a local network anyway, and I don't know what else to say about this.

If anyone knows a better way for storing client secrets on a public client, please do educate me (BTW, these are temporary tokens).

Reply:
> GitHub however requires you to send the client secret while auth flow to be sent in the URL parameters 😱🤯, which can be easily intercepted on a local network anyway

That is false, as HTTPS encrypts the path, query and fragment of the URL (the scheme is also technically encrypted, but the port number 443 with TLS encrypted payload tends to give away that it’s encrypted HTTPS traffic)

[The only part that is currently not encrypted in most cases is the hostname (the “api.github.com” part), but there’s work to encrypt that too.](https://en.wikipedia.org/wiki/Server_Name_Indication)

Reply:
> > GitHub however requires you to send the client secret while auth flow to be sent in the URL parameters 😱🤯, which can be easily intercepted on a local network anyway
>
> HTTPS encrypts the path, query and fragment of the URL

I'm pretty sure I was drunk when I wrote that (or late at night) 😅. I stand corrected!